lib/spack/spack/cmd/list.py

```diff
@@ -126,6 +127,31 @@ def rows_for_ncols(elts, ncols):
         yield row
 
 
+@formatter
+def version_json(pkg_names, out):
+    """..."""
+    pkgs = [spack.repo.get(name) for name in pkg_names]
+
+    out.write('[\n')
+
+    # output name and latest version for each package
+    pkg_latest = ",\n".join(
+        ['  {{"name": "{0}",\n'
+         '   "latest_version": "{1}",\n'
+         '   "homepage": "{2}",\n'
+         '   "file": "{3}"'
+         '}}'.format(
+              pkg.name,
+              VersionList(pkg.versions).highest(),
```

Comment:
this is also returning "infinity_versions" such as `develop` as soon as they are present... ideally, we should return what `spack install <package>` would select, e.g. the latest numbered version if present.